### PR TITLE
CLion better docs

### PR DESCRIPTION
**Pull request by lasote**

### CODE REVIEW

File: integrations/clion.rst [hunk 1]
```diff
@@ -13,17 +13,199 @@ Just include the ``conanbuildinfo.cmake`` this way:
...
+
+2. Edit the ``CMakeLists.txt`` file and add the following lines:
+
+.. code-block:: cmake
+
+   if(EXISTS ${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+       include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+       conan_basic_setup()
+   else()
+       message(WARNING "Skipping Conan")
```
Comment (memsharded) [inline integrations/clion.rst]: I'd say to use a better message, like "conanbuildinfo.cmake doesn't exist, you have to run ``conan install``", or something similar.

File: integrations/clion.rst [hunk 2]
```diff
@@ -13,17 +13,199 @@ Just include the ``conanbuildinfo.cmake`` this way:
...
+
+   $ cd cmake-build-debug
+   $ conan install ../ -s build_type=Debug
+
+6. Repeat the last step if you have the release build types configured in your CLion IDE, but changing the build_type
+setting accordingly:
+
+.. code-block:: bash
+
+   $ cd cmake-build-debug
```
Comment (memsharded) [inline integrations/clion.rst]: cd cmake-build-release?

File: integrations/clion.rst [hunk 3]
```diff
@@ -13,17 +13,199 @@ Just include the ``conanbuildinfo.cmake`` this way:
...
+
+1. Create a new CLion project
+
+|wizard_new|
+
+2. Edit the ``CMakeLists.txt`` file and add the following lines:
+
+.. code-block:: cmake
+
+   if(EXISTS ${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
```
Comment (memsharded) [inline integrations/clion.rst]: Maybe provide a repo, with a full working example.

File: integrations/clion.rst [hunk 4]
```diff
@@ -13,17 +13,199 @@ Just include the ``conanbuildinfo.cmake`` this way:
...
```
Comment (memsharded) [inline integrations/clion.rst]: Maybe provide a full repo, with a real "Hello world" or "Poco" or real "Zlib usage"

File: integrations/clion.rst [hunk 5]
```diff
@@ -13,17 +13,199 @@ Just include the ``conanbuildinfo.cmake`` this way:
...
+
+
+4. To build your library with CLion follow the guide of :ref:`Using packages from the step 5<step_five>`.
+
+5. To package your library use the ``conan package`` command passing the used build folder:
+
+.. code-block:: bash
+
+   $ mkdir package && cd package
+   $ conan package ../ --build_folder=../cmake-build-debug
```
Comment (memsharded) [inline integrations/clion.rst]: I think this will only work when released 0.23.

File: integrations/clion.rst [hunk 6]
```diff
@@ -13,17 +13,199 @@ Just include the ``conanbuildinfo.cmake`` this way:
...
+- A ``lib`` folder containing our library
+- A ``include`` folder containing our header files
+- A ``conaninfo.txt`` and a	``conanmanifest.txt`` conan files.
+
+7. If everything looks good in your ``package`` folder you can export your recipe to your local cache and/or upload it
+to a conan server.
+
+.. code-block:: bash
+
+   $ conan export myuser/channel
```
Comment (memsharded) [inline integrations/clion.rst]: Make more evident that this only uploads the recipe, not the binaries

File: integrations/clion.rst (hunk 4 above)
Comment (lasote) [inline integrations/clion.rst]: I prefer to keep the clion hello world library example, I will edit the cpp and introduce some lines, I think it's clearer to show how to create your library.

File: integrations/clion.rst (hunk 5 above)
Comment (lasote) [inline integrations/clion.rst]: Changing to develop

File: integrations/clion.rst (hunk 4 above)
Comment (memsharded) [inline integrations/clion.rst]: Ok, good

**Pull request merged**